lighthouse-core/lib/page-functions.js

```diff
@@ -257,6 +268,8 @@ function getNodePath(node) {
 /**
  * @param {Element} node
  * @return {string}
+ *
+ * Note: if the node resides within shadow DOM, the selector only starts from the root
```

Comment:
what is root? rootElement of the main document?

Reply:
it is the shadow root i think, as in the selector would be relative from the shadow root rather than from the main document. but would be nice to specify that if its the case, maybe paul can confirm

Reply:
added more comments around this.